scheduler/pkg/envoy/processor/incremental_test.go

```diff
@@ -687,291 +855,309 @@ func TestEnvoySettings(t *testing.T) {
-			},
-			numExpectedClusters: 4,
-			numExpectedRoutes:   1,
-			numTrafficSplits:    map[string]int{"model": 2},
-		},
-		{
-			name: "Rolling update complete",
-			ops: []func(inc *IncrementalProcessor, g *WithT){
-				createTestServer("server", 1),
-				createTestModel("model", "server", 1, []int{0}, 1, []store.ModelReplicaState{store.Available}),
-				createTestModel("model", "server", 1, []int{0}, 2, []store.ModelReplicaState{store.Available}),
-			},
-			numExpectedClusters: 2,
-			numExpectedRoutes:   1,
-			numTrafficSplits:    map[string]int{"model": 1},
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			modelStore := store.NewMemoryStore(log.New(), store.NewLocalSchedulerStore(), nil)
-			inc := &IncrementalProcessor{
-				cache:            cache.NewSnapshotCache(false, cache.IDHash{}, log.New()),
-				logger:           log.New(),
-				xdsCache:         xdscache.NewSeldonXDSCache(log.New(), &xdscache.PipelineGatewayDetails{Host: "pipeline", GrpcPort: 1, HttpPort: 2}),
-				modelStore:       modelStore,
-				experimentServer: experiment.NewExperimentServer(log.New(), nil, nil, nil),
-				pipelineHandler:  pipeline.NewPipelineStore(log.New(), nil, modelStore),
-			}
-			inc.xdsCache.AddListeners()
-			for _, op := range test.ops {
-				op(inc, g)
+			data, err := os.ReadFile("snapshots/" + routeFilename)
+			g.Expect(err).To(BeNil())
+
+			var rawMessages []json.RawMessage
+			err = json.Unmarshal(data, &rawMessages)
+			g.Expect(err).To(BeNil())
+
+			count := 0
+			for _, rawMessage := range rawMessages {
+				snapshotRouteConfig := &routev3.RouteConfiguration{}
+				err := protojson.Unmarshal(rawMessage, snapshotRouteConfig)
+				g.Expect(err).To(BeNil())
+
+				resultingRouteConfig := resultingRoutes[snapshotRouteConfig.Name]
+				g.Expect(resultingRouteConfig).To(Not(BeNil()))
+				g.Expect(len(snapshotRouteConfig.VirtualHosts)).Should(Equal(1))
+				g.Expect(len(resultingRouteConfig.VirtualHosts)).Should(Equal(1))
+				snapshotRoutes := getTrafficSplits(snapshotRouteConfig.VirtualHosts[0])
+				resultingRoutes := getTrafficSplits(resultingRouteConfig.VirtualHosts[0])
+				g.Expect(len(resultingRoutes)).Should(Equal(len(snapshotRoutes)))
+				g.Expect(resultingRoutes).Should(ConsistOf(snapshotRoutes))
+				count++
 			}
-			g.Expect(len(inc.xdsCache.Clusters)).To(Equal(test.numExpectedClusters))
-			g.Expect(len(inc.xdsCache.Routes)).To(Equal(test.numExpectedRoutes))
-			for modelName, trafficSplits := range test.numTrafficSplits {
-				g.Expect(len(inc.xdsCache.Routes[modelName].Clusters)).To(Equal(trafficSplits))
+			g.Expect(len(resultingRoutes)).To(Equal(count))
+
+			clusterFilename := test.snapshotFilename + "-clusters.json"
+			if *generateSnapshots {
+				createSnapshot(g, inc.xdsCache.ClusterContents(), clusterFilename)
 			}
 
+			resultingClusters := getResultingClusters(inc.xdsCache.ClusterContents())
+
+			data, err = os.ReadFile("snapshots/" + clusterFilename)
```

Comment:
extract as constant for the directory name

Reply:
done